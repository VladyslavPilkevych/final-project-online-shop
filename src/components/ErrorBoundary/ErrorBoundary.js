import React, { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import styles from './ErrorBoundary.module.scss';
import serverErrImg from '../../assets/images/ErrorBoundary/serverErr.jpg';
import Button from '../Button/Button';

function ErrorBoundary() {
  const navigate = useNavigate();
  const [showError, setShowError] = useState(false);
  const newErrorFromRedux = useSelector((state) => state.products.errorBoundaryString);
  useEffect(() => {
    if (newErrorFromRedux) {
      setShowError(true);
    }
  }, [newErrorFromRedux]);
  const reloadPage = () => {
    window.location.reload();
    setShowError(false);
  };
  const goToHomePage = () => {
    navigate({
      pathname: '/',
    });
    reloadPage();
    setShowError(false);
  };
  return (
    <div>
      {showError
        && (
          <div className={styles.errorWrapper}>
            <h1 className={styles.errorMainText}>
              Sorry, there are some problems with our server.
            </h1>
            <h2 className={styles.errorMessage}>
              Error message:
              {' '}
              {newErrorFromRedux.message}
            </h2>
            <img className={styles.errorImage} src={serverErrImg} alt="serverErrImg" />
            <div className={styles.errorBtns}>
              <Button className={styles.errorBtn} handleClick={goToHomePage}>
                Return to Home page
              </Button>
              <Button className={styles.errorBtn} handleClick={reloadPage}>
                Reload this page
              </Button>
            </div>
          </div>
        )}
    </div>
  );
}

export default ErrorBoundary;