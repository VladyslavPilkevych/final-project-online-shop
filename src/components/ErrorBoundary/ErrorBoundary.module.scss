@import '../../assets/styles/global.module.scss';

.errorWrapper {
  height: 100%;
  width: 100%;
  position: fixed;
  overflow-y: scroll;
  z-index: 999;
  background-color: white;
  margin: 0;
  text-align: center;
  justify-content: center;
}

.errorMainText {
  font-family: $fonts;
  font-weight: 600;
  font-size: 38px;
  margin: 20px 0;
}

.errorMessage {
  font-family: $fonts;
  font-weight: 400;
  font-size: 25px;
}

.errorImage {
  margin: 20px 0;
  max-width: 70%;
  max-height: 60%;
}

.errorBtns {
  display: block;
  margin: 0 auto;
}

.errorBtn {
  padding: 10px;
  margin: 5px;
  border-radius: 50px;
  background-color: $blue;
  color: white;
  font-family: $fonts;
  font-weight: 400;
  font-size: 18px;
}