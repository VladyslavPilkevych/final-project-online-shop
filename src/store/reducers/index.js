import { combineReducers } from 'redux';
import preloaderReducer from './preloaderReducer';
import productsReducer from './productsReducer';
import menuReducer from './menuReducer';
import userReducer from './userReducer';
import searchReducer from './searchReducer';
import cartReducer from './cartReducer';
import miniMenuReducer from './miniMenuReducer';
import filtersCategoriesReducer from './filtersCategoriesReducer';

const reducer = combineReducers({
  loader: preloaderReducer,
  products: productsReducer,
  menu: menuReducer,
  miniMenu: miniMenuReducer,
  filtersCategories: filtersCategoriesReducer,
  user: userReducer,
  search: searchReducer,
  cart: cartReducer,
});

export default reducer;
