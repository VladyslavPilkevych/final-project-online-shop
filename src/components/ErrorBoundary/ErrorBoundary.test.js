import { render } from '@testing-library/react';
import ErrorBoundary from './ErrorBoundary';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../store';

function Component() {
  return (
    <BrowserRouter>
      <Provider store={store}>
        <ErrorBoundary />
      </Provider>
    </BrowserRouter>
  );
}

describe('Snapshot ErrorBoundary should render', () => {
  test('Snapshot ErrorBoundary is rendered', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});