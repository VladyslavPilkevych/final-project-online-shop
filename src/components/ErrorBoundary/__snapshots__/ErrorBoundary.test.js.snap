// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot ErrorBoundary should render Snapshot ErrorBoundary is rendered 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
